ompi/mca/coll/base/coll_base_scatter.c

```diff
@@ -61,14 +61,15 @@
  */
 int
 ompi_coll_base_scatter_intra_binomial(
-    const void *sbuf, int scount, struct ompi_datatype_t *sdtype,
-    void *rbuf, int rcount, struct ompi_datatype_t *rdtype,
+    const void *sbuf, size_t scount, struct ompi_datatype_t *sdtype,
+    void *rbuf, size_t rcount, struct ompi_datatype_t *rdtype,
     int root, struct ompi_communicator_t *comm,
     mca_coll_base_module_t *module)
 {
     mca_coll_base_module_t *base_module = (mca_coll_base_module_t*)module;
     mca_coll_base_comm_t *data = base_module->base_data;
-    int line = -1, rank, vrank, size, err, packed_size, curr_count;
+    int line = -1, rank, vrank, size, err, packed_size;
```

Comment:
We should probably get rid of the `packed_sizet` below and make `packed_size` `size_t`.